diagram/diagram_test.go

```diff
@@ -0,0 +1,437 @@
+package diagram
```

Comment:
So... I don't think there's a problem with importing the diagram code into core_test. So instead of duplicating all the test code here, could we just change the function that sets up the mock transfer so that optionally includes the state diagrammer components if some environment variable is set?